### PR TITLE
Ver 2.0 of Azure Diags Script

### PR DESCRIPTION
**Pull request by JimGBritt**

### CODE REVIEW

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 1]
```diff
@@ -763,12 +913,18 @@ $JSONRULES = @'
         }
 '@
 }
-$JSONVar = @'
+if(!($EXportInitiative))
```
Comment (rwsanders) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1]: needs consistent casing in variable name

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 2]
```diff
@@ -992,13 +1237,18 @@ $JSONRULES = @'
             }
         }
 '@
-
-$JSONVar = @'
+if(!($EXportInitiative))
```
Comment (rwsanders) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1]: needs consistent casing in variable name

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 3]
```diff
@@ -1414,55 +1798,155 @@ IF($($ExportEH) -or ($ExportLA))
...
+                    $PolicyJSON = Update-EventHubJSON -resourceType $Type.ResourceType -metricsArray $metricsArray -logsArray $logsArray -nameField $RPVar[1] -JSONType $JSONType -ExportInitiative $ExportInitiative -kind $Type.Kind -PolicyResourceDisplayName $PolicyResourceDisplayName -PolicyName $ShortNameRT
+                    $PolicyBag = $PolicyBag + $PolicyJSON[2] +  ',' + "`r`n"
+                    $PolicyDefParam = @'
+                {
+                    "policyDefinitionId": <PolicyResoureceID>,
+                    <Policy Definition Parameters>
+                },
+'@
+                $PolicyDefParam = $PolicyDefParam.Replace("<PolicyResoureceID>", "$($PolicyRSID)")
+                $PolicyDefParam = $PolicyDefParam.Replace("<Policy Definition Parameters>", "$($PolicyJSON[3])")
```
Comment (rwsanders) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1919]: I'm having trouble finding where PolicyJSON[3] is instantiated and assigned.  I believe this is the root cause of my failure in attempting the following command.  If this is intended to be parameters should this be PolicyJSON[0]?
`.\Create-AzDiagPolicy.ps1 -ExportAll -ExportEH -ValidateJSON -ExportDir ".\LogPolicies3" -SubscriptionId "<subscriptionId>" -AllRegions -ExportInitiative`

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 4]
```diff
@@ -1414,55 +1798,155 @@ IF($($ExportEH) -or ($ExportLA))
...
                     {
-                        $NULL = new-item -ItemType Directory -Path "$($ExportDir)\$($RPVar[0])"
-                    }
-                    catch
+                        "policyDefinitionId": <PolicyResoureceID>,
+                        <Policy Definition Parameters>
+                    },
+'@
+                    $PolicyDefParam = $PolicyDefParam.Replace("<PolicyResoureceID>", "$($PolicyRSID)")
+                    $PolicyDefParam = $PolicyDefParam.Replace("<Policy Definition Parameters>", "$($PolicyJSON[3])")
```
Comment (rwsanders) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: I'm having trouble finding where PolicyJSON[3] is instantiated and assigned.  I believe this is the root cause of my failure in attempting the following command.  If this is intended to be parameters should this be PolicyJSON[0]?
`.\Create-AzDiagPolicy.ps1 -ExportAll -ExportLA -ValidateJSON -ExportDir ".\LogPolicies3" -SubscriptionId "<subscriptionId>" -AllRegions -ExportInitiative`

**Review by rwsanders — COMMENTED** [review]
This is a really cool tool and is already very solid. I would like to use this script to generate diagnostics policies on a regular cadence across the board or when we detect changes to log categories. 

I did have an issue using the ExportInitiative parameter and pointed out what I think may be the root cause.

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 (hunk 4 above)
Comment (rwsanders) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: tested a little more the PolicyJSON[0] was not it... but there is a "parameters" property that is missing from the output.  This causes the Format-JSON function (a slick function btw) to fail.  I was able to test this by statically defining `'"parameters":{}'` in place of `$($PolicyJSON[3])`.

Comment (JimGBritt) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: Thank you! I'll dig into this Rob.  I appreciate it!  Stay tuned.

Comment (JimGBritt) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: I know what the issue is.  All regions missed in Log Analytics logic. That is my bad.  Resolution coming soon.  Thank you!

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 (hunk 1 above)
Comment (JimGBritt) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1]: fixed thank you! ;)

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 (hunk 4 above)
Comment (JimGBritt) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: This is fixed with the latest commit, please review Rob @rwsanders

Comment (rwsanders) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1846]: great it exports the templates and the templates deploy.  thanks!

**Pull request merged**